OrbitSshQt/IntegrationTests.cpp

```diff
@@ -173,50 +174,121 @@ TEST(OrbitSshQtTests, IntegrationTest) {
       });
 
   QObject::connect(&sftp_channel, &OrbitSshQt::SftpChannel::errorOccurred,
-                   &loop, [&](std::error_code e) {
-                     loop.quit();
+                   loop.get(), [&](std::error_code e) {
+                     loop->quit();
                      FAIL() << absl::StrFormat(
                          "SFTP channel error occurred: %s", e.message());
                    });
 
-  QObject::connect(&sftp_channel, &OrbitSshQt::SftpChannel::stopped, &loop,
+  QObject::connect(&sftp_channel, &OrbitSshQt::SftpChannel::stopped, loop.get(),
                    [&]() {
                      LOG("Sftp channel closed!");
-                     loop.quit();
+                     loop->quit();
                      CheckCheckpoint(Checkpoint::kSftpChannelStopped);
                    });
 
   // SFTP Operation
 
   QObject::connect(&sftp_op,
-                   &OrbitSshQt::SftpCopyToRemoteOperation::errorOccurred, &loop,
-                   [&](std::error_code e) {
-                     loop.quit();
+                   &OrbitSshQt::SftpCopyToRemoteOperation::errorOccurred,
+                   loop.get(), [&](std::error_code e) {
+                     loop->quit();
                      FAIL() << absl::StrFormat(
                          "SFTP operation error occurred: %s", e.message());
                    });
 
   QObject::connect(&sftp_op, &OrbitSshQt::SftpCopyToRemoteOperation::stopped,
-                   &loop, [&]() {
+                   loop.get(), [&]() {
                      LOG("Sftp file copy finished!");
                      sftp_channel.Stop();
                      CheckCheckpoint(Checkpoint::kSftpOperationStopped);
                    });
 
   session.ConnectToServer(creds);
   LOG("connect to server");
-  QTimer::singleShot(std::chrono::seconds{5}, [&]() {
-    loop.quit();
-    FAIL()
-        << "Timeout occurred. The whole integration test should be done in 5 "
-           "seconds. If not, probably it's stuck somewhere in the callback "
-           "logic.";
+  QTimer::singleShot(std::chrono::seconds{5}, [loop]() {
```

Comment:
The bug you mentioned could have also been fixed by using another overload of QTimer::singleShot where you specify a receiving object: `QTimer::singleShot(std::chrono::seconds{5}, &loop, [&]() {...`

Then the lambda will only be executed when loop still exists.

Reply:
done. ptal

Reply:
Thanks. Looks good!